framework/packages/abstract-std/src/native/ibc.rs

```diff
@@ -15,29 +15,24 @@ use crate::{
 #[cosmwasm_schema::cw_serde]
 // ANCHOR: callback-info
 pub struct CallbackInfo {
-    /// Used to identify the callback that is sent (acts like the reply ID)
-    pub id: String,
     /// Used to add information to the callback.
     /// This is usually used to provide information to the ibc callback function for context
-    pub msg: Option<Binary>,
+    pub payload: Binary,
```

Comment:
Why did you opt to use payload here? It's a name that isn't really used anywhere else while "msg" indicates that it's a message type, and should be deserialized as such.

Reply:
Because payload can be anything. And msg can be confused with initiator_msg that we have in the Executor case

Reply:
How can it be anything? We want to set conventions and imo we should encourage the use of `CallbackMsg` as the convention

Reply:
Pretty sure it have more usecases than just being a message. cosmwasm_std::Reply also use `payload` in this case https://docs.rs/cosmwasm-std/2.0.4/cosmwasm_std/struct.Reply.html#structfield.payload

Reply:
On replies you still have the REPLY_ID that is used to uniquely identify replies. If we don't set the soft standard to use messages then it could be very easy to use it incorrectly.

If both of you agree then its fine the way it is. 

I do think we could rename `CallbackInfo` as well. It's kind of a weird name, would prefer just `Callback`